### PR TITLE
Add AST transformations to match our JavaScript API

### PR DESCRIPTION
**Pull request by freezy**
This adds two "transformers": A transformer is a class that takes in an AST and spits out a modified version of it. Name up to discussion. ;)

It also adds a method to the table object that will transpile and execute the table script.

## Scope Transformer

Since in Visual Pinball's VBScript, everything is global, we need to scope stuff. What the scope transformer does is take the whole script and pack it into a function that then takes in arguments which contain the objects the script will interact with. Currently this is done only for the playfield objects, but in the future there will also be VPScript's standard library and probably others.

So given the AST of this script:

```js
BallRelease.CreateBall();
```

will result in an AST of this script:

```js
window.initTableScript = items => {
    items.BallRelease.CreateBall();
}
```

Note that this is only applied to identifiers that actually exist as table items.

## Event Transformer

VBScript has this weird notion of defining event callbacks by defining a `Sub` with the name of the object suffixed by the event name. For example, the callback of the `Init` event of a `Plunger` object looks like that:

```vbs
Sub Plunger_Init
   ' do something here
EndSub
```

This won't obviously work in JavaScript. What we do instead is inherit from `EventEmitter` and emit events on the object directly. So the event transformer rewrites the AST resulting in:

```js
Plunger.on('Init', () => {
   // do something here
});
```
This is applied, if:
- the object with the given name actually exists
- the object supports the given event

## Transpiler

The transpiler is the class parsing the VBS, creating the AST, applying the transformers, generating the JavaScript and executing it.

## Other Changes

- The parsing unit tests were renamed to `post-processor.spec.ts`. 
- The helper used to compile the VBS (`vbsToJs()`) has moved to the test folder since it's only used there.
- `estree.ts` got a few type updates
- Dependencies were bumped

### CODE REVIEW

File: lib/scripting/event-transformer.spec.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	const three = new ThreeHelper();
+	let table: Table;
+
+	before(async () => {
+		table = await Table.load(new NodeBinaryReader(three.fixturePath('table-gate.vpx')));
+	});
+
+	it('should transform a valid event on a valid item', () => {
+
+		const vbs = `Sub WireRectangle_Init()\n    BallRelease.CreateBall\nEnd Sub\n`;
```
Comment (jsm174) [inline lib/scripting/event-transformer.spec.ts]: With the upcoming changes to the grammar, we are removing the whitespace checks from the beginning of block statements. I think these tests will fail.

**Comment by freezy** [comment]
Thanks! Is it okay if I merge and let you rebase and fix the tests according to your changes?

**Comment by jsm174** [comment]
Sounds like a plan!

**Comment by freezy** [comment]
Thanks!

**Pull request merged**